applogic/src/api/messages.rs

```diff
@@ -198,3 +199,50 @@ impl User {
             .unwrap_or_default()
     }
 }
+
+pub(crate) fn group_messages(messages: Vec<ConversationMessage>) -> Vec<UiConversationMessage> {
+    let mut grouped_messages = Vec::new();
+    let mut iter = messages.into_iter().peekable();
+
+    while let Some(conversation_message) = iter.next() {
+        let mut timestamp = conversation_message.timestamp();
+        match conversation_message.message() {
```

Comment:
This should probably be a `let Message::Content(content_message) = ... else { ... }`.